pkg/agent/agent.go

```diff
@@ -216,14 +216,6 @@ func (k *Klocksmith) watchUpdateStatus(update func(s updateengine.Status)) {
 
 	go k.ue.ReceiveStatuses(ch, stop)
 
-	// synthesize first message
```

Comment:
Isn't the idea of `c.ch = make(chan *dbus.Signal, signalBuffer)` and `c.conn.Signal(c.ch)` that `c.ch` receives all signal messages and thus, so does `ReceiveStatus`, which converts that message to a Status and sends it to the `ch` we pass?

If not and we only expect one status, I don't see a reason to loop over a channel, which I think is what you're saying be unrolling it in option 2.